src/sentry/models/integrations/integration.py

```diff
@@ -105,14 +105,22 @@ def add_organization(self, organization: RpcOrganization, user=None, default_aut
         from sentry.models import OrganizationIntegration
 
         try:
-            org_integration, created = OrganizationIntegration.objects.get_or_create(
-                organization_id=organization.id,
-                integration_id=self.id,
-                defaults={"default_auth_id": default_auth_id, "config": {}},
-            )
-            # TODO(Steve): add audit log if created
-            if not created and default_auth_id:
-                org_integration.update(default_auth_id=default_auth_id)
+            with transaction.atomic():
+                org_integration, created = OrganizationIntegration.objects.get_or_create(
+                    organization_id=organization.id,
+                    integration_id=self.id,
+                    defaults={"default_auth_id": default_auth_id, "config": {}},
+                )
+                # TODO(Steve): add audit log if created
+                if not created and default_auth_id:
+                    org_integration.update(default_auth_id=default_auth_id)
+
+                organization_service.schedule_signal(
```

Comment:
@corps hmm shouldn't you send this signal then created is true?

Reply:
👍 Can do.